src/Overlay.tsx

```diff
@@ -205,6 +205,7 @@ function Overlay({
             ...overlayProps,
             placement,
             show,
+            className: !transition && show && 'show',
```

Comment:
```suggestion
            ...(!transition && show && { className: 'show' }),
```

This change will avoid overwriting `className` when transition is true and spreading props after a className is declared.

Reply:
`overlayProps` won't contain the `className` if `overlay` is a function? The `className` will probably be hard-coded in the function.

Reply:
There's no `className` in the `overlayProps`.  This is what's passed back from react-overlays:

https://github.com/react-bootstrap/react-overlays/blob/f9f1a14752a8d8ff9aa7c7d207aa0c413ca70660/src/Overlay.tsx#L117-L130

The above suggestion will prevent the below `className="abc"` from being overwritten when `transition=true`.
```
(props) => <Tooltip className="abc" {...props}>my tooltip</Tooltip
```